DQM/SiPixelPhase1Summary/src/SiPixelPhase1Summary.cc

```diff
@@ -152,30 +197,103 @@ void SiPixelPhase1Summary::fillSummaries(DQMStore::IBooker & iBooker, DQMStore::
 	  continue; // Ignore non-existing MEs, as this can cause the whole thing to crash
 	}
 
-	if (me->hasError()) {
-	  //If there is an error, fill with 0
-	  summaryMap_[name]->setBinContent(i+1,j+1,0);
-	} //Do we want to include warnings here?
-	else if (me->hasWarning()){
-	  summaryMap_[name]->setBinContent(i+1,j+1,0.5);
+	if (!summaryMap_[name]){
+	  edm::LogWarning("SiPixelPhase1Summary") << "Summary map " << name << " is not available !!";
+	  continue; // Based on reported errors it seems possible that we're trying to access a non-existant summary map, so if the map doesn't exist but we're trying to access it here we'll skip it instead.
 	}
-	else summaryMap_[name]->setBinContent(i+1,j+1,1);
+	if ((me->getQReports()).size()!=0) summaryMap_[name]->setBinContent(i+1,j+1,(me->getQReports())[0]->getQTresult());
+	else summaryMap_[name]->setBinContent(i+1,j+1,-1);
       }  
     }    
   }
   //Now we will use the other summary maps to create the overall map.
   for (int i = 0; i < 12; i++){ // !??!?!? xAxisLabels_.size() ?!?!
+    if (!summaryMap_["Grand"]){
+      edm::LogWarning("SiPixelPhase1Summary") << "Grand summary does not exist!";
+      break;
+    }
     for (int j = 0; j < 4; j++){ // !??!?!? yAxisLabels_.size() ?!?!?!
       summaryMap_["Grand"]->setBinContent(i+1,j+1,1); // This resets the map to be good. We only then set it to 0 if there has been a problem in one of the other summaries.
       for (auto const mapInfo: summaryPlotName_){ //Check summary maps
 	auto name = mapInfo.first;
 	if (name == "Grand") continue;
-	if (summaryMap_[name]->getBinContent(i+1,j+1) < 0.9 && summaryMap_["Grand"]->getBinContent(i+1,j+1) > summaryMap_[name]->getBinContent(i+1,j+1)) summaryMap_["Grand"]->setBinContent(i+1,j+1,summaryMap_[name]->getBinContent(i+1,j+1)); // This could be changed to include warnings if we want?
+	if (!summaryMap_[name]){
+	  edm::LogWarning("SiPixelPhase1Summary") << "Summary " << name << " does not exist!";
+	  continue;
+	}
+	if (summaryMap_["Grand"]->getBinContent(i+1,j+1) > summaryMap_[name]->getBinContent(i+1,j+1)) summaryMap_["Grand"]->setBinContent(i+1,j+1,summaryMap_[name]->getBinContent(i+1,j+1));
       }
     }
   }
 
 }
 
+//------------------------------------------------------------------
+// Fill the trend plots
+//------------------------------------------------------------------
+void SiPixelPhase1Summary::fillTrendPlots(DQMStore::IBooker & iBooker, DQMStore::IGetter & iGetter, int lumiSec){
+
+  // If we're running in online mode and the lumi section is not modulo 10, return. Offline running always uses lumiSec=0, so it will pass this test.
```

Comment:
I don't really understand this comment. why is lumiSec=0 always? (lumiSec seems to be defined as lumiSeg.id().luminosityBlock();